### PR TITLE
fix(remoting): fix forgetting doAfterRPCHooks in Async

### PR DESCRIPTION
**Pull request by FantasyChen-DH**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

fix #<xxx> 

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by echooymxq** [comment]
close #6379

File: remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingAbstract.java [hunk 1]
```diff
@@ -355,19 +355,22 @@ private Runnable buildProcessRequestHandler(ChannelHandlerContext ctx, RemotingC
...
+
             if (responseFuture.getInvokeCallback() != null) {
                 executeInvokeCallback(responseFuture);
             } else {
                 responseFuture.putResponse(cmd);
                 responseFuture.release();
             }
         } else {
-            log.warn("receive response, but not matched any request, " + RemotingHelper.parseChannelRemoteAddr(ctx.channel()));
+            log.warn("receive response, but not matched any request, " + remoteAddr);
```
Comment (aaron-ai) [inline remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingAbstract.java:373]:
```suggestion
            log.warn("receive response, but not matched any request, remote address={}", remoteAddr);
```

File: remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingAbstract.java [hunk 2]
```diff
@@ -355,19 +355,22 @@ private Runnable buildProcessRequestHandler(ChannelHandlerContext ctx, RemotingC
     public void processResponseCommand(ChannelHandlerContext ctx, RemotingCommand cmd) {
         final int opaque = cmd.getOpaque();
         final ResponseFuture responseFuture = responseTable.get(opaque);
+        final String remoteAddr = ctx == null ? "" : RemotingHelper.parseChannelRemoteAddr(ctx.channel());
         if (responseFuture != null) {
             responseFuture.setResponseCommand(cmd);
 
             responseTable.remove(opaque);
 
+            doAfterRpcHooks(remoteAddr, responseFuture.getRequestCommand(), cmd);
```
Comment (keranbingaa) [inline remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingAbstract.java:364]: maybe invokesync will call this method twice?

Comment (keranbingaa) [reply remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingAbstract.java:364]: I mean in func invokesync of NettyRemotingClient, the doAfterRpcHooks is called now, if you call doAfterRpcHooks in this place, may the doAfterRpcHooks will be called twice?

Comment (FantasyChen-DH) [reply remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingAbstract.java:364]: u r right,  I didn't consider that aspect. So we should delete the doAfterRpcHooks in func invokeSync.

**Pull request closed without merging**